src/services.ts

```diff
@@ -0,0 +1,154 @@
+import { Context } from "hono";
+import { Env } from "./env";
+import { Service, ServiceAuthType } from "./types";
+
+class Services {
```

Comment:
See comment in env.ts. 

If we can have dynamic environment keys then I don't think this is even necessary.

Reply:
If I make the environment variables dynamic and eliminate the services class, how can I obtain the remaining necessary elements such as host, parameter/header names? Would that be provided by the client in headers?

Reply:
Please let me know if you even want it to be stored for fallback, or if you completely want to rely on headers.